mitodl/hugo-course-publisher
# remove automatic search behavior

#### Pre-Flight checklist


#### What are the relevant tickets?

none

#### What's this PR do?

This PR removes the automatic search behavior, which would basically wait 300ms until the user had finished typing and then fire off a search request. Instead, the user now has to press enter or click the little spyglass in order to run the search.

I also added the latest version of our eslint rules, and fixed issues found after the upgrade.

#### How should this be manually tested?

check out the search page and make sure it feels good and works as you would expect.
